performancecopilot/pcp
# qa, selinux, small code changes

Changes committed to git@github.com:kmcdonell/pcp.git 20200201

Ken McDonell (11):
      qa/1362: tweaks for Ubuntu/Debian
      selinux: assorted minor fixups
      qa/admin/other-packages: update check-vm rules for redis on Ubuntu
      build/rpm/pcp.spec.in: fix guard for
      build/rpm/pcp.spec.in: fix guard for /usr/bin/hostname
      qa/admin/other-packages/unavailable: add line for Debian bullseye/sid
      qa/1185: refactor to use sample PMDA metrics
      qa/1343: add df diagnostic on the fail path
      src/python/pmapi.c: remove unused variable
      qa/common.check: be more careful of crontab updates
      qa/188: knock-on from last openSUSE change ... need small rework for SLES

 build/rpm/pcp.spec.in               |    4 ++--
 qa/1185                             |    6 +++---
 qa/1185.out                         |   29 ++++++++++++++++++++---------
 qa/1343                             |    1 +
 qa/1362                             |   19 ++++++++++++++++++-
 qa/1622                             |    9 +++++++++
 qa/188                              |    4 +---
 qa/917.out.in                       |    2 +-
 qa/admin/other-packages/manifest    |    2 +-
 qa/admin/other-packages/unavailable |    9 +++++----
 qa/common.check                     |   13 ++++++++++++-
 src/python/pmapi.c                  |    1 -
 src/selinux/README                  |    2 +-
 src/selinux/pcpupstream.te.in       |    2 +-
 14 files changed, 75 insertions(+), 28 deletions(-)

Details ...

commit 43482c6eab674884b0075b7705d8c57156586594
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sun Feb 2 11:50:22 2020 +1100

    qa/188: knock-on from last openSUSE change ... need small rework for SLES

commit 73b2df1d25367d393282bfb670e20eb5ad193293
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sun Feb 2 11:49:03 2020 +1100

    qa/common.check: be more careful of crontab updates
    
    Need to cull the comments to avoid polluting the crontab with
    3 comments for every time qa/255 is run.

commit 1e366feb49f8c525dd86fb8bd5a2faa45316b6a9
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sun Feb 2 07:09:47 2020 +1100

    src/python/pmapi.c: remove unused variable
    
    sts in pmUnits_int() ... observed by gcc on vm22 (F29).

commit f1e3a9220c56196c06e76cad189aed13ba2c064d
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 21:04:54 2020 +1100

    qa/1343: add df diagnostic on the fail path

commit c0164a24d6d9ef33865cf698466b1c35974253d5
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 21:03:29 2020 +1100

    qa/1185: refactor to use sample PMDA metrics
    
    Get away from the platform PMDAs that have non-deterministic
    output across the platforms.

commit d5ef5432fe72a5225b156fc070571f70753a6496
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 17:43:23 2020 +1100

    qa/admin/other-packages/unavailable: add line for Debian bullseye/sid

commit a6193352205a3ed0c87b4f787f77ac7ce41359a3
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 17:39:22 2020 +1100

    build/rpm/pcp.spec.in: fix guard for /usr/bin/hostname
    
    Was busted in multiple ways ... build failed on vm14 CentOS 6.10 for
    example.

commit e07d860c41bae29c090743e5c7c391400080d603
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 17:38:03 2020 +1100

    build/rpm/pcp.spec.in: fix guard for

commit c5b5349fc99f357d240b5216a02479a4ad803deb
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 16:37:25 2020 +1100

    qa/admin/other-packages: update check-vm rules for redis on Ubuntu
    
    Package name seems to have changed over time for the client side.
    
    These changes are incomplete ... more similar changes will be needed
    as I stumble across other dpkg-based systems.

commit dc21263a32a08e5ef9fd2f366dedda64e6cc9d1b
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 16:33:15 2020 +1100

    selinux: assorted minor fixups
    
    - add ioctl to list of allowed ops in the pmda-smart block for
      fixed_disk_device_t:blk_file in the pcpupstream.te.in file
    - update qa/917.out.in to match
    - add some missing cases (not just the one above) to qa/1622
    - update the README to stress that both qa 917 and 1622 need to
      be reviewed after pcpupstream.te.in changes

commit 3ce1fca60aff27be9f768e730b5dd93a1e0aa367
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sat Feb 1 16:30:13 2020 +1100

    qa/1362: tweaks for Ubuntu/Debian
    
    - must use systemctl, not service, to force postgresql restart
    - so add _notrun if systemctl not installed
    - update comment re. postgresql major version upgrade steps for
      Ubuntu (and all Debian-based distros)
